source/manual/aws-iam-key-rotation.html.md

```diff
@@ -51,6 +51,10 @@ Keys may also be shared across AWS and Carrenza environments.
 
 If the key is not present in the hieradata, it might be present in the control panel of one of the 3rd party services, such as Fastly or Logit.
 
+> **Note**
+> If the key is not present in `govuk-secrets` nor in any other 3rd party services,
+> you can [remove the old key](#5-remove-the-old-key)
```

Comment:
If this relates to the key not being in use, then it might be that the user needs removing, not just the key right?

Reply:
The user may have a new key and the old key is no longer present

Reply:
Right, so this relates to the limitation of having up to 2 keys per user right? If you've got two keys, and one is unused, you'll need to remove that unused key prior to creating a new key.

If that is the case, then I guess you don't need to check if it's present anywhere right? You can just trust the AWS interface when it tells you it's not in use?

Reply:
For what I understand about AWS users and keys, this doesn't only relate to the case where a user already has 2 keys. There could also be users with only one key and that key could be not in use.

In that case, I'm not sure if we can just trust the AWS interface when it tells us it's not in use. Isn't it possible that we have that key in puppet but not have used it?

Reply:
> For what I understand about AWS users and keys, this doesn't only relate to the case where a user already has 2 keys. There could also be users with only one key and that key could be not in use.
> 
> In that case, I'm not sure if we can just trust the AWS interface when it tells us it's not in use. Isn't it possible that we have that key in puppet but not have used it?

In the case where the single acesss key isn't being used, we should be able to trust the AWS interface, but it might be good to check govuk-secrets as well just to see if there's redundant configuration that can be removed.

In this scenario, ideally one would figure out if the user is still relevant, or redundant. It looks like there are users without keys that are being used, so unfortunately it's not clear cut.

I don't know how far we want to go with these docs, but I think it's wise to be quite specific.

Reply:
> I don't know how far we want to go with these docs, but I think it's wise to be quite specific.

I agree. Is there a particular change you have in mind?

Reply:
@cbaines unless there's something specific you meant by "I think it's wise to be quite specific", are you happy to move this forward?

Reply:
We can trust the user interface to tell us if a key is in use. If a key is in puppet and not in use then there's a good chance that the key should have been removed as part of a refactor but wasn't.

I think that If we find an unused/irrelevant user then it should be removed. But maybe instead the users could be added to a list of users to investigate and remove?

Reply:
> @cbaines unless there's something specific you meant by "I think it's wise to be quite specific", are you happy to move this forward?

I can't add any clarity as I don't know what the intent is here (apart from "Update the docs...").

I'm OK with merging this change, but at least when I read it, I'm not sure when that additional conditional instruction to remove a key would apply. I can't work out if it's truly a useful side note, or something more important that should be included in the main block of text.